client/dashboard/sites/settings-caching/index.tsx

```diff
@@ -0,0 +1,234 @@
+		}
+		handleClearObjectCache();
+
+		setIsClearingAllCaches( true );
+	};
+
+	const renderCallout = () => {
+		return <SettingsCallout siteSlug={ siteSlug } />;
+	};
+
+	const renderForm = () => {
+		return (
+			<>
+				<Card>
+					<CardBody>
+						<form onSubmit={ handleUpdateEdgeCacheStatus }>
+							<VStack spacing={ 4 } style={ { padding: '8px 0' } }>
+								<DataForm< CachingFormData >
+									data={ formData }
+									fields={ fields }
+									form={ form }
+									onChange={ ( edits: Partial< CachingFormData > ) => {
+										setFormData( ( data ) => ( { ...data, ...edits } ) );
+									} }
+								/>
+								<HStack justify="flex-start">
+									<Button
+										variant="primary"
+										type="submit"
+										isBusy={ isPending }
+										disabled={ isPending || ! isDirty }
+									>
+										{ __( 'Save' ) }
+									</Button>
+								</HStack>
+							</VStack>
+						</form>
+					</CardBody>
+				</Card>
+
+				<VStack spacing={ 4 }>
+					<ActionList
+						title={ __( 'Clear caches' ) }
+						description={ __(
+							'Clearing the cache may temporarily make your site less responsive.'
+						) }
+					>
+						<ActionList.ActionItem
+							title={ __( 'Global edge cache' ) }
+							description={ __( 'Edge caching enables faster content delivery.' ) }
+							actions={
+								<Button
+									variant="secondary"
+									size="compact"
+									onClick={ handleClearEdgeCache }
+									isBusy={ edgeCacheClearMutation.isPending && ! isClearingAllCaches }
+									disabled={
+										! isEdgeCacheActive || edgeCacheClearMutation.isPending || isClearingAllCaches
+									}
+								>
+									{ __( 'Clear' ) }
+								</Button>
+							}
+						/>
+						<ActionList.ActionItem
+							title={ __( 'Object cache' ) }
+							description={ __( 'Data is cached using Memcached to reduce database lookups.' ) }
+							actions={
+								<Button
+									variant="secondary"
+									size="compact"
+									onClick={ handleClearObjectCache }
+									isBusy={ objectCacheClearMutation.isPending && ! isClearingAllCaches }
+									disabled={ objectCacheClearMutation.isPending || isClearingAllCaches }
+								>
+									{ __( 'Clear' ) }
+								</Button>
+							}
+						/>
+					</ActionList>
+					<ActionList>
+						<ActionList.ActionItem
+							title={ __( 'Clear all caches' ) }
+							actions={
+								<Button
+									variant="secondary"
+									size="compact"
+									onClick={ handleClearAllCaches }
+									isBusy={ isClearingAllCaches }
+									disabled={ isClearingAllCaches }
+								>
+									{ __( 'Clear all' ) }
+								</Button>
+							}
+						/>
+					</ActionList>
+				</VStack>
+			</>
+		);
+	};
+
+	const description = canUpdate
+		? createInterpolateElement(
+				__( 'Manage your site’s server-side caching. <learnMoreLink />.' ),
+				{
+					learnMoreLink: <a href="#learn-more">{ __( 'Learn more' ) }</a>,
+				}
+		  )
+		: '';
```

Comment:
There's also a description for the upsell 🙂

![Screenshot 2025-06-02 at 4 22 31 PM](https://github.com/user-attachments/assets/a334f560-c2ba-4549-8124-5fcffa777b0a)